ts/header.ts

```diff
@@ -0,0 +1,49 @@
+import { ButtonAction } from './enums.js';
+
+export const renderHeader = (date: Date, updateDate: (newDate: Date) => void) => {
```

Comment:
I guess updateDate function type is repeated quite a few times, since it is passed as param. so could be typed in types.ts and reused